recipe/meta.yaml

```
@@ -202,6 +204,18 @@ outputs:
         # ensure only one test is left (namely, our hint)
         - python -c "import sys; lines=open('testlog').readlines(); sys.exit(0 if any('======== 1 failed' in x for x in lines) else 1)"
 
+  - name: scipy-typed
+    requirements:
+      host:
+        - python
+      run:
+        - python
+        - {{ pin_subpackage('scipy', exact=True) }}
+        - scipy-stubs >={{ version }}.0,<{{ next_micro_version }}
```

> This PR pins the exact version so that the stubs installed are compatible with the SciPy version.

Does `scipy-stubs` always have a matching version to scipy? In that case, we could simplify this to
```suggestion
        - scipy-stubs =={{ version }}.*
```
(and get rid of the extra jinja variables)

---

Yea, it uses the supported `scipy` version (major.minor.patch) as a prefix. So currently it's at `1.14.1.5`, matching `scipy == 1.14.1`. The last version number (`5` in this case) is incremental and specific to `scipy-stubs`.

---

Matching exactly on patch/micro version is almost certainly going to be annoying at some point (exact matches will go missing as soon as @jorenham is a bit late with a new release). It's also highly unlikely that there will be issues with patch version mismatches, since we don't add new functionality or change signatures in patch versions.

I'd go with `<next_minor_version`, so we get the latest patch version from `scipy.stubs`.

---

> exact matches will go missing as soon as @jorenham is a bit late with a new release

For minor and major releases there'll be an RC period where I can even release `scipy-stubs` *before* `scipy` (as I plan to do with 1.15.0). But as for `scipy` patch releases, it'll be trickier, but not impossible. Because if I know beforehand when the scipy patch release will be published, it should be possible to release a matching `scipy-stubs` version justs before that, which should be "valid enough" if the API changes will be minimal.

---

> it'll be trickier, but not impossible:

It's not 100% impossible, but it is indeed tricky - and there is no real reason to require it. I assume you do take a holiday sometimes at least, so why would we add a pin so exact that if a new `scipy-stubs` 1.xx.yy release is a day late then `scipy-types` completely fails to install?

---

Yea you're right. And in case of patch releases `scipy-stubs` will for the most part 
already be compatible anyway.